### PR TITLE
add message buffer implementation

### LINKED ISSUE #27: Update Messaging System for DAQ MainTask to Queue or Buffer
From Pull Request

I implemented a shitty circular buffer somewhere in the FC FW ([here](https://github.com/UBC-Rocket/Whistler-Blackcomb-v2/blob/master/common/src/buffer.c)). Take a look and see if it would be useful to port over here, unless of course there is some non-shitty solution you know of and are already planning on using.

_Originally posted by @Esouder in https://github.com/UBC-Rocket/WB-DAQ-Firmware/pull/26#discussion_r877788731_

### PR DESCRIPTION
**Pull request by Amarantheum**
This adds a simple circular buffer implementation for storing messages. This fixes #27

### CODE REVIEW

**Comment by Amarantheum** [comment]
Tested this on the physical DAQ board with a 10 second delay in the main task and it was able to execute the inputted commands.

File: source/WB-CAN-Firmware.c
Comment (Esouder) [inline source/WB-CAN-Firmware.c]: This is purely stylistic, but I think C-style camelCase would be better for all of variable names, etc. (throughout the PR) than snake case, to match the rest of the project.

File: source/WB-CAN-Firmware.c
Comment (Esouder) [inline source/WB-CAN-Firmware.c:229]: Good catch to get rid of this `semaphore_message` I agree that it's not needed anymore. The lock on the buffer is performing this role now, right?

File: source/WB-CAN-Firmware.c
Comment (Esouder) [inline source/WB-CAN-Firmware.c]: Good question. @davinbirdi?

File: source/lib/buffer.h
Comment (Esouder) [inline source/lib/buffer.h:18]: If we end up doing this with any reasonable frequency, we can add some functions to check this to this file. Otherwise this is a great precaution to have, I think.

File: source/lib/message_types.h
Comment (Esouder) [inline source/lib/message_types.h:12]: Someday this might get reorganized into a different file when we get around to implementing whatever CAN protocol we come up with.

**Review by Esouder — APPROVED** [review]
Looks great! Only major (and yet still very minor) thing is to change to camelCase. Please get Davin's approval as well before merging. 

I've resolved a couple comments immediately after reviewing since they were for information/commenting only.

File: source/WB-CAN-Firmware.c
Comment (davinbirdi) [reply source/WB-CAN-Firmware.c]: Lower camelcase for functions and variables: 

https://github.com/UBC-Rocket/Whistler-Blackcomb-v2/blob/726de347c98f1fe272c6d635334212baebc70326/doc/WritingCleanCode.md

Comment (davinbirdi) [reply source/WB-CAN-Firmware.c]: I do like the snake case in this case.

File: source/WB-CAN-Firmware.c
Comment (davinbirdi) [reply source/WB-CAN-Firmware.c:229]: 👍

File: source/WB-CAN-Firmware.c
Comment (davinbirdi) [reply source/WB-CAN-Firmware.c]: In the line under the comment, there is a RTT Print that prints out a bunch of data to the RTT Terminal which is logged as a CSV file.

I was printing out whatever value was in the `message` variable to let me know when a certain message (DPR Resume or Pause) was received, so when looking at the CSV file to plot the input signal and output signal I would know when to start looking at the data when DPR is actually on vs sensor data being read when the board is just on.

Comment (davinbirdi) [reply source/WB-CAN-Firmware.c]: Justin commented a ToDo in another commit - so this is resolved for now until we develop a better way to log data.

File: source/WB-CAN-Firmware.c
Comment (davinbirdi) [inline source/WB-CAN-Firmware.c:168]: This is great. The reason we did this was to allow our main control task to reference this task using this handle, specifically for Suspending and Resuming Tasks.

Line 241:
`vTaskSuspend(actuator_task);`

File: source/WB-CAN-Firmware.c
Comment (davinbirdi) [inline source/WB-CAN-Firmware.c:440]: This is a billion times better and exactly what we want.

File: source/WB-CAN-Firmware.c
Comment (davinbirdi) [inline source/WB-CAN-Firmware.c:390]: Let's quickly fix this so it doesn't cause issues next time we run the program.

**Review by davinbirdi — APPROVED** [review]
Great implementation, liking the coding style and well-commented code.

File: source/WB-CAN-Firmware.c
Comment (davinbirdi) [reply source/WB-CAN-Firmware.c:390]: Done

**Review by davinbirdi — APPROVED** [review]

**Pull request merged**
